flyspell-correct.el

```diff
@@ -123,8 +123,18 @@ of (command, word) to be used by `flyspell-do-correct'."
          (skip "[SKIP]")
          (result (completing-read
                   (format "Correcting '%s': " word)
-                  (append candidates
-                          (list save accept-session accept-buffer skip)))))
+                  ;; Use function with metadata to disable sorting.
+                  (lambda (input predicate action)
```

Comment:
Didn't know that it's possible to use function as a collection for `completing-read`. TIL! More information in [manual][1].

  [1]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Programmed-Completion.html#Programmed-Completion